src/ctia/http/routes/common.clj

```diff
@@ -57,6 +57,33 @@
    (s/optional-key :search_after) (describe [s/Str] "Pagination stateless cursor")
    (s/optional-key :limit) (describe Long "Pagination Limit")})
 
+(s/defn prep-sort_by-param-schema :- (s/protocol s/Schema)
+  "Generalize sort_by enum schema from (enum fields...) to s/Str
+  so it can accept the `field1:asc,field2:desc` syntax to sort by
+  multiple fields. Add the original enums to the doc for the route,
+  that also explains how this syntax works."
+  [search-q-params]
+  (cond-> search-q-params
+    (st/get-in search-q-params [:sort_by])
+    (st/update :sort_by
+               (fn [sort_by-enum]
```

Comment:
Would doing an `s/fn` and expressing the type in the param work as an alternative the `assert` below?

Reply:
I like @marioaquino's idea too.

I wonder if there's a combinator we could have to make those `_` bindings go away in a more graceful way where a nice alternative isn't obvious. Something that takes the rhs of the binding and an assertion predicate.

```clojure
(let [valid-fields (chitty!! (:vs sort_by-enum) #(every? (some-fn string? simple-ident?) %))
      ...]
  ...)
```

But `s/fn` is grand if it covers the case.

Reply:
(It could be either-monad polymorphic, in case @yogsototh's work makes it further into the codebase.)

Reply:
My intention here was to communicate to any poor dev in the future that violates these invariants as directly as possible. An assert with a custom msg seems to work.

Reply:
@j0ni Maybe something like this, but I prefer the `_` version.

```clojure
(let [valid-fields (doto (:vs sort_by-enum)
                     (as-> valid-fields (assert (some-fn string? simple-ident?) valid-fields) (pr-str valid-fields)
...
```